src/modules/goals/goal.service.ts

```diff
@@ -0,0 +1,113 @@
+import { Injectable } from '@nestjs/common';
+import { Goal, GoalStatus } from '@prisma/client';
+import { PrismaService } from '../../prisma.service';
+import { GetMetricsQueryDTO } from '../metrics/dto';
+import { MetricsService } from '../metrics/metrics.service';
+import { GetGoalsQueryDto } from './dto/get-goals-query.dto';
+import { GoalDTO } from './dto/goal.dto';
+
+@Injectable()
+export class GoalService {
+  constructor(
+    private prismaService: PrismaService,
+    private metricsService: MetricsService
+  ) {}
+
+  async createGoal(goal: GoalDTO): Promise<Goal> {
+    const createdGoal = await this.prismaService.goal.create({ data: goal });
+    return createdGoal;
+  }
+
+  async getGoalById(id: number): Promise<Goal | null> {
+    const goal = await this.prismaService.goal.findUnique({
+      where: { id },
+    });
+
+    if (!goal) {
+      return null;
+    }
+      
+    return this.checkGoalStatus(goal);
+  }
+
+  async findAll(): Promise<Goal[]> {
+    const updatedGoals = await this.prismaService.goal.findMany({  
+      orderBy:  { id: 'asc' },
+    }).then(goals => Promise.all(goals.map(goal => this.checkGoalStatus(goal))));
```

Comment:
En vez del .then, podes hacer `return Promise.all(goals.map(goal => this.checkGoalStatus(goal))`